src/main/scala/com/redhat/et/silex/statistics/cramersv.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * cramersv.scala
+ * author:  RJ Nowling <rnowling@redhat.com>
+ *
+ * Copyright (c) 2016 Red Hat, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.et.silex.statistics
+
+import scala.util.Random
+
+/**
+ * "[[https://en.wikipedia.org/wiki/Cram%C3%A9r%27s_V Cramers' V]] is a measure of
+ * association between two nominal variables, giving a value between 0 and +1
+ * (inclusive)." 
+ */
+object CramersV {
+  /**
+   * Calculate Cramer's V for two sets (`values1` and `values2`) of co-sampled values.
+   */
+  def apply[T, U](values1 : Seq[T], values2 : Seq[U]) : Double = {
+    val set1 = values1.toSet
+    val set2 = values2.toSet
+
+    if (set1.size == 1 && set2.size == 1) {
+      1.0
+    } else if (values1.size == 0 || set1.size == 1 || set2.size == 1) {
+      0.0
+    } else {
+      val pairCounts = values1.zip(values2)
```

Comment:
Does this function assume values1.length == values2.length?   (if so that should be checked).  Note that `zip` will not error out if the lengths are different.
Similarly, does it assume set1.size == set2.size?